client/state/themes/selectors/get-theme-signup-url.js

```diff
@@ -19,19 +20,41 @@ export function getThemeSignupUrl( state, themeId, options = {} ) {
 
 	const { tabFilter, styleVariationSlug } = options;
 	const themeType = getThemeType( state, themeId );
+
+	let themeTypeParam = themeType;
+	// Map theme tier values to the values expected by the signup flow.
+	const { themeTier } = getThemeTier( state, null, themeId );
+	// If there is no themeTier then there's nothing to map (dot-org themes don't have tiers)
+	if ( config.isEnabled( 'themes/tiers' ) && themeTier?.slug ) {
```

Comment:
[Nitpick] I think it would be more readable to remove the `themeTier?.slug` check and include it in the switch with a new case for when we don't have a slug.

Reply:
Added it to the clean-up list as something to consider when cleaning up the feature flag. 👍